### PR TITLE
Implemented Add link to functionality.

### PR DESCRIPTION
**Pull request by jd-alexander**
Gutenberg Mobile PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/2810
WP Android PR https://github.com/wordpress-mobile/WordPress-Android/pull/13405

## Description

This PR adds the `Link To` functionality in the File Block settings. When you select this option you will be greeted with a modal that allows you to select either an Attachment File or Media File as the link of the File Block. Doing so will add the appropriate `<a href>` tag to the block. 

## How has this been tested?

To properly test this behavior you will need to use one of the parent apps as the `getMedia` function requires the file to exist within the site. So if you test this PR here, you will see the URL as the only Link To option as seen below.

<kbd></kbd>

Once you have this PR's submodule reference updated in the respective parent app project, you can follow the testing instructions below. 

#### Media File

1. Create a new post. 
2. Add a File Block. 
3. Add a file. 
4. Click the settings button. 
5. Ensure the Link To is set to Media File. 
6. Close this modal. 
7. Switch to HTML mode. 
8. Go to the File Block within the html.
9. Observe that the href within the first a tag has a link to the media file. To be sure, you will see the file's extension here.
 
<kbd></kbd>

#### Attachment Page

1. Create a new post. 
2. Add a File Block. 
3. Add a file. 
4. Click the settings button. 
5. Ensure the Link To is set to Attachment Page. 
6. Close this modal. 
7. Switch to HTML mode. 
8. Go to the File Block within the html.
9. Observe that the the href within the first a tag has a link to a WordPress page. 

<kbd></kbd>

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/file/edit.native.js [hunk 1]
```diff
@@ -175,27 +191,40 @@ export class FileEdit extends Component {
...
 		);
 
 		const isCopyUrlDisabled = isUploadFailed || isUploadInProgress;
-		const dimmedStyle = isCopyUrlDisabled && styles.disabledButton;
-		const finalButtonStyle = Object.assign(
-			{},
-			actionButtonStyle,
-			dimmedStyle
-		);
-
```
Comment (etoledom) [inline packages/block-library/src/file/edit.native.js:194]: This is breaking the disabled state style (dimmed 25%)
Probably a merge error?

Comment (jd-alexander) [reply packages/block-library/src/file/edit.native.js:194]: Thanks for spotting. I will check it out. 🙏🏾

Comment (jd-alexander) [reply packages/block-library/src/file/edit.native.js]: Resolved in ff743fb

**Comment by jd-alexander** [comment]
@etoledom this is ready for another review 🙇

**Comment by etoledom** [comment]
Working great now, thanks for the update!

I left a comment [here](https://github.com/wordpress-mobile/WordPress-Android/pull/13405#issuecomment-730258875), but since the code looks good, I'm thinking about merging this one and tackling the issue on a separated PR.

What do you think?

**Comment by jd-alexander** [comment]
> I left a comment here, but since the code looks good, I'm thinking about merging this one and tackling the issue on a separated PR.

> What do you think?

Yes, that sounds like a good idea for sure!

**Review by etoledom — APPROVED** [review]
As we agreed, let's go ahead with this.
Great job @jd-alexander !

**Pull request merged**